src/Compilers/Core/Portable/SourceGenerator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    public interface ISourceGenerator
+    {
+        void Execute(SourceGeneratorContext context);
+    }
+
+    public struct SourceGeneratorContext
+    {
+        public Compilation Compilation { get; }
+
+        // TODO: replace AnalyzerOptions with an differently named type that is otherwise identical.
```

Comment:
AnalyzerOptions is the type given to an analyzer to get access to the additional files and .editorconfig settings. We _could_ just use the same type here, the TODO is tracking that we should probably have something else only because future growth of the APIs might take the types in different directions.